src/test/cypress/support/commands.ts

```diff
@@ -140,3 +141,15 @@ Cypress.Commands.add('getSettled', (selector, opts = {}) => {
         });
     });
 });
+
+/**
+ * Servers that use bamboo and bitbucket need a sleep between creating a course and creating a programming exercise for group synchronization.
+ * This command sleeps if necessary (determined by the value in the configuration).
+ * */
+Cypress.Commands.add('waitForGroupSynchronization', () => {
+    const runsOnBamboo: boolean = Cypress.env('isBamboo');
+    if (runsOnBamboo) {
```

Comment:
I have a question here: couldn't we check for the flag `runsOnBamboo` to be true before calling `waitForGroupSynchronization`? I posted a comment, where I would think you could add it below.
I think the method name is a bit misleading, because if `runsOnBamboo` is `false` we do _not_ wait. The bare method call in the other file indicates, that we wait here. It's just a thought, maybe you find it reasonable :)

Reply:
Yeah we can, but then every code segment, which might have to do group synchronization has to add an extra if statement. The idea was that the whole decision logic whether or not the test needs to sleep for group synchronization is put into the `waitForGroupSynchronization` command.  Generally the rule is if the tests run on Bamboo, so when `runsOnBamboo = true` we have to wait for the group synchronization.
The crux is that group synchronization only has to be waited for once. So if a test creates a course and then three programming exercises the group synchonization sleep only has to occur **once**. This is why I added the boolean flag below, to override the group synchronization check